extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -590,51 +590,73 @@ export class Bind {
     });
   }
 
+
+  /**
+   * Determines which properties to update based on results of evaluation
+   * of all bound expression strings with the current scope. This method
+   * will only return properties that need to be updated along with their
+   * new value.
+   * @param {!Array<!BoundPropertyDef>} boundProperties
+   * @param {Object<string, ./bind-expression.BindExpressionResultDef>} results
+   * @return {
+   *   !Array<{
+   *     boundProperty: !BoundPropertyDef,
+   *     newValue: !./bind-expression.BindExpressionResultDef,
+   *   }>
+   * }
+   * @private
+   */
+  calculateUpdates_(boundProperties, results) {
+    const propertyUpdates = [];
+    boundProperties.forEach(boundProperty => {
+      const {expressionString, previousResult} = boundProperty;
+      const newValue = results[expressionString];
+      if (newValue === undefined ||
+          this.shallowEquals_(newValue, previousResult)) {
+        user().fine(TAG, `Expression result unchanged or missing: ` +
+            `"${expressionString}"`);
+        return;
+      }
+
+      boundProperty.previousResult = newValue;
+      user().fine(TAG, `New expression result: ` +
+          `"${expressionString}" -> ${newValue}`);
+      propertyUpdates.push({boundProperty, newValue});
```

Comment:
Optional: Now that this is a smaller block of code, move this into an `else` block of the conditional above and remove the `return`.